src/snowflake/connector/sf_dirs.py

```diff
@@ -31,12 +31,12 @@ def _resolve_platform_dirs() -> PlatformDirsProto:
         "appname": "snowflake",
         "appauthor": False,
     }
-    snowflake_home = os.path.expanduser(
+    snowflake_home = pathlib.Path(
         os.environ.get("SNOWFLAKE_HOME", "~/.snowflake/"),
-    )
-    if os.path.exists(snowflake_home):
```

Comment:
I'm just curious is there a bug related to the previous implementation?

I checked the code and think they are effectively equivalent.. or it's more about code improvement?

```python
import os
import pathlib

assert pathlib.Path(os.path.expanduser("~/.snowflake/")) == pathlib.Path(str(pathlib.Path("~/.snowflake/").expanduser()))

```

Reply:
Ahhh, I see - I didn't notice that this is handled via `pathlib` later on. I think this might be worth adding anyway, as the code becomes less confusing (I honestly thought that there was a bug in there), although the connector works properly.
I've updated the PR description and description.md